spec/integration/rails_spec.rb

```diff
@@ -12,41 +12,31 @@
   require 'action_controller/railtie'
   require 'action_mailer/railtie'
 
-  # We don't use :mock_intake, as we want the stubs to stay around between
-  # individual examples
-  RSpec.describe 'Rails integration', :allow_running_agent, :spec_logger do
+  RSpec.describe 'Rails integration' do
     include Rack::Test::Methods
-    include MockIntake::WaitFor
+    include_context 'event_collector'
 
-    def app
-      @app ||= Rails.application
-    end
-
-    after :each do
-      MockIntake.clear!
+    let(:app) do
+      Rails.application
     end
 
     after :all do
       ElasticAPM.stop
+      ElasticAPM::Transport::Worker.adapter = nil
```

Comment:
I feel like this and the `TestAdapter` ought to be part of `EventCollector` so it could be more easily shared between multiple specs.

Reply:
good idea!